### PR TITLE
Replace deprecated functions and fix compilation warnings

### PR DESCRIPTION
**Pull request by zhuyaliang**
Replacing GtkAction with GAction
Replacing GtkUIManager with GtkBuilder

### CODE REVIEW

**Comment by lukefromdc** [comment]
I will have to check this when I get home to ensure we don't lose any icons in menus with this

File: src/fr-window.c [hunk 1]
```diff
@@ -5661,94 +5643,83 @@ fr_window_construct (FrWindow *window)
...
-			  G_CALLBACK (activate_action_open),
-			  window);
-	gtk_action_group_add_action (actions, action);
-	g_object_unref (action);
+	window->priv->actions = actions = (GActionGroup*)g_simple_action_group_new ();
 
 	/* other actions */
+	g_action_map_add_action_entries (G_ACTION_MAP (actions),
+                                   action_entries, G_N_ELEMENTS (action_entries),
+                                   window);
```
Comment (raveit65) [inline src/fr-window.c:5653]: Indentation

File: src/fr-window.c [hunk 2]
```diff
@@ -5661,94 +5643,83 @@ fr_window_construct (FrWindow *window)
...
 		g_error_free (error);
 	}
+    /*Add shortcut keys*/
+	for (int i = 0; i < G_N_ELEMENTS (menu_keybindings); i++)
+	{
+		menuitem = (GtkWidget*)gtk_builder_get_object (ui, menu_keybindings[i].widget_id);
+		gtk_widget_add_accelerator (menuitem, "activate", accel_group,
+                                menu_keybindings[i].keyval,
+                                menu_keybindings[i].modifier,
+                                GTK_ACCEL_VISIBLE);
```
Comment (raveit65) [inline src/fr-window.c:5675]: Indentation

File: src/fr-window.c [hunk 3]
```diff
@@ -5661,94 +5643,83 @@ fr_window_construct (FrWindow *window)
...
+	}
+
+	gtk_builder_add_callback_symbols (ui,
+                                   "menu_item_select_cb", G_CALLBACK (menu_item_select_cb),
+                                   "menu_item_deselect_cb", G_CALLBACK (menu_item_deselect_cb),
+                                    NULL);
```
Comment (raveit65) [inline src/fr-window.c:5681]: Indentation

File: src/fr-window.c [hunk 4]
```diff
@@ -5125,31 +5111,35 @@ recent_chooser_item_activated_cb (GtkRecentChooser *chooser,
...
 
-	g_signal_connect (G_OBJECT (chooser),
-			  "item_activated",
-			  G_CALLBACK (recent_chooser_item_activated_cb),
-			  window);
+	g_signal_connect (G_OBJECT (menu),
+                     "item_activated",
+                      G_CALLBACK (recent_chooser_item_activated_cb),
+                      window);
+
```
Comment (raveit65) [inline src/fr-window.c:5141]: Indentation and one unneeded row.

File: src/fr-window.c [hunk 5]
```diff
@@ -5661,94 +5643,83 @@ fr_window_construct (FrWindow *window)
...
+	g_action_map_add_action (G_ACTION_MAP (actions), G_ACTION (action));
+	g_object_unref (action);
 
-	menubar = gtk_ui_manager_get_widget (ui, "/MenuBar");
+	menu_radio_files = (GtkWidget *)gtk_builder_get_object (ui, "view_all_files_item");
+	g_signal_connect (G_OBJECT (menu_radio_files),
+                   "toggled",
+                    G_CALLBACK (view_all_files_selected_toggled),
+                    window);
```
Comment (raveit65) [inline src/fr-window.c:5706]: Indentation

**Comment by raveit65** [comment]
New build warnings which aren't in previous build.
https://app.travis-ci.com/github/mate-desktop/engrampa/jobs/601515415#L1191
```
In file included from fr-window.c:54:

./ui.h:30:48: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "New", activate_action_new, NULL, NULL, NULL},

                                                      ^

./ui.h:31:50: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Open", activate_action_open, NULL, NULL, NULL},

                                                        ^

./ui.h:32:55: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "SaveAs", activate_action_save_as, NULL, NULL, NULL},

                                                             ^

./ui.h:33:56: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Extract", activate_action_extract, NULL, NULL, NULL},

                                                              ^

./ui.h:34:65: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "TestArchive", activate_action_test_archive, NULL, NULL, NULL},

                                                                       ^

./ui.h:35:63: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Properties",  activate_action_properties, NULL, NULL, NULL},

                                                                     ^

./ui.h:36:52: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Close", activate_action_close, NULL, NULL, NULL},

                                                          ^

./ui.h:38:50: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Copy", activate_action_copy, NULL, NULL, NULL},

                                                        ^

./ui.h:39:48: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Cut", activate_action_cut, NULL, NULL, NULL},

                                                      ^

./ui.h:40:52: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Paste", activate_action_paste, NULL, NULL, NULL},

                                                          ^

./ui.h:41:54: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Rename", activate_action_rename, NULL, NULL, NULL},

                                                            ^

./ui.h:42:54: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Delete", activate_action_delete, NULL, NULL, NULL},

                                                            ^

./ui.h:43:61: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "SelectAll", activate_action_select_all, NULL, NULL, NULL},

                                                                   ^

./ui.h:44:65: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "DeselectAll", activate_action_deselect_all, NULL, NULL, NULL},

                                                                       ^

./ui.h:45:50: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Find", activate_action_find, NULL, NULL, NULL},

                                                        ^

./ui.h:46:59: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "AddFiles", activate_action_add_files, NULL, NULL, NULL},

                                                                 ^

./ui.h:47:61: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "AddFolder", activate_action_add_folder, NULL, NULL, NULL},

                                                                   ^

./ui.h:48:58: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Password", activate_action_password, NULL, NULL, NULL},

                                                                ^

./ui.h:50:78: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "ViewToolbar", activate_toggle, NULL, "true", activate_action_view_toolbar},

                                                                                    ^

./ui.h:51:82: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "ViewStatusbar", activate_toggle, NULL, "true", activate_action_view_statusbar},

                                                                                        ^

./ui.h:52:80: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "ViewFolders",  activate_toggle, NULL, "false", activate_action_view_folders},

                                                                                      ^

./ui.h:53:63: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "LastOutput", activate_action_last_output, NULL, NULL, NULL},

                                                                     ^

./ui.h:54:50: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Stop", activate_action_stop, NULL, NULL, NULL},

                                                        ^

./ui.h:55:54: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Reload", activate_action_reload, NULL, NULL, NULL},

                                                            ^

./ui.h:57:52: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "About", activate_action_about, NULL, NULL, NULL},

                                                          ^

./ui.h:58:56: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "Contents", activate_action_manual, NULL, NULL, NULL},

                                                              ^

./ui.h:60:64: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "OpenRecent_Toolbar", activate_action_open, NULL, NULL, NULL},

                                                                      ^

./ui.h:61:63: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "ExtractToolbar", activate_action_extract, NULL, NULL, NULL},

                                                                     ^

./ui.h:62:66: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "AddFilesToolbar", activate_action_add_files, NULL, NULL, NULL},

                                                                        ^

./ui.h:63:68: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "AddFolderToolbar", activate_action_add_folder, NULL, NULL, NULL},

                                                                          ^

./ui.h:65:55: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "GoBack", activate_action_go_back, NULL, NULL, NULL},

                                                             ^

./ui.h:66:61: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "GoForward", activate_action_go_forward, NULL, NULL, NULL},

                                                                   ^

./ui.h:67:51: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "GoUp", activate_action_go_up, NULL, NULL, NULL},

                                                         ^

./ui.h:68:55: warning: missing field 'padding' initializer [-Wmissing-field-initializers]

        { "GoHome", activate_action_go_home, NULL, NULL, NULL},
```

File: src/fr-window.c [hunk 6]
```diff
@@ -5661,94 +5643,83 @@ fr_window_construct (FrWindow *window)
...
+	menuitem = (GtkWidget *)gtk_builder_get_object (ui, "open_recent_item");
+	gtk_menu_item_set_submenu (GTK_MENU_ITEM (menuitem), window->priv->menubar_recentmenu);
+
+    /* open recent toolbar item action  */
+	menuitem = (GtkWidget *)gtk_builder_get_object (ui, "tool_open_recent_item");
+	window->priv->toolbar_recentmenu = fr_window_create_recent_chooser (window);
+	gtk_menu_tool_button_set_menu (GTK_MENU_TOOL_BUTTON (menuitem), window->priv->toolbar_recentmenu);
+
+	if (window->priv->list_mode == FR_WINDOW_LIST_MODE_FLAT)
+		action = g_simple_action_new_stateful ("ViewAllFiles", G_VARIANT_TYPE_STRING, g_variant_new_string ("files"));
```
Comment (raveit65) [inline src/fr-window.c:5695]: New build warning.
https://app.travis-ci.com/github/mate-desktop/engrampa/jobs/601515415#L1439
```
fr-window.c:5695:58: warning: Casting a non-structure type to a structure type and accessing a field can lead to memory access errors or data corruption [alpha.core.CastToStruct]

                action = g_simple_action_new_stateful ("ViewAllFiles", G_VARIANT_TYPE_STRING, g_variant_new_string ("files"));

                                                                       ^~~~~~~~~~~~~~~~~~~~~

/usr/include/glib-2.0/glib/gvarianttype.h:124:46: note: expanded from macro 'G_VARIANT_TYPE_STRING'

#define G_VARIANT_TYPE_STRING               ((const GVariantType *) "s")

                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~

fr-window.c:5697:58: warning: Casting a non-structure type to a structure type and accessing a field can lead to memory access errors or data corruption [alpha.core.CastToStruct]

                action = g_simple_action_new_stateful ("ViewAllFiles", G_VARIANT_TYPE_STRING, g_variant_new_string ("folder"));

                                                                       ^~~~~~~~~~~~~~~~~~~~~

/usr/include/glib-2.0/glib/gvarianttype.h:124:46: note: expanded from macro 'G_VARIANT_TYPE_STRING'

#define G_VARIANT_TYPE_STRING               ((const GVariantType *) "s")

                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: src/fr-window.c (hunk 1 above)
Comment (zhuyaliang) [reply src/fr-window.c]: done

File: src/fr-window.c (hunk 2 above)
Comment (zhuyaliang) [reply src/fr-window.c]: done

File: src/fr-window.c (hunk 6 above)
Comment (zhuyaliang) [reply src/fr-window.c:5694]: This warning can be retained,I used ```G_VARIANT_TYPE_STRING```
```
#define G_VARIANT_TYPE_STRING               ((const GVariantType *) "s")
```

**Comment by zhuyaliang** [comment]
@raveit65  Fix all warnings

**Comment by raveit65** [comment]
It seems that all icons are there in menus but we get a runtime warning.
```
(engrampa:71925): Gtk-CRITICAL **: 15:54:22.343: gtk_widget_hide: assertion 'GTK_IS_WIDGET (widget)' failed
```

**Comment by zhuyaliang** [comment]
> It seems that all icons are there in menus but we get a runtime warning.
> 
> ```
> (engrampa:71925): Gtk-CRITICAL **: 15:54:22.343: gtk_widget_hide: assertion 'GTK_IS_WIDGET (widget)' failed
> ```

Warning fixed

**Review by lukefromdc — APPROVED** [review]
This seems to work, I noticed no change in menu behavior and all expected icons showed up. Manually setting up menu items using GtkImageMenuItem seems to work fine here, and later we can port that over to MateImageMenuItem (from mate-desktop 1.27.1 or later). That can be in a later PR, this makes ready for it-and if we later get bugs from this, it will be easier to bisect if these are two separate commits

**Review by raveit65 — APPROVED** [review]
LGTM,
builds and runs fine.

**Pull request merged**